src/utils/IdealGraphVisualizer/FilterWindow/src/main/java/com/sun/hotspot/igv/filterwindow/FilterTopComponent.java

```diff
@@ -327,80 +342,46 @@ private static String getJsHelperText() {
     }
 
-    private FilterTopComponent() {
-        filterSettingsChangedEvent = new ChangedEvent<>(this);
-        initComponents();
-        setName(NbBundle.getMessage(FilterTopComponent.class, "CTL_FilterTopComponent"));
-        setToolTipText(NbBundle.getMessage(FilterTopComponent.class, "HINT_FilterTopComponent"));
-        //        setIcon(Utilities.loadImage(ICON_PATH, true));
-
-        sequence = new FilterChain();
-        filterChain = new FilterChain();
-        ScriptEngineManager sem = new ScriptEngineManager();
-        engine = sem.getEngineByName("ECMAScript");
-        try {
-            engine.eval(getJsHelperText());
-        } catch (ScriptException ex) {
-            Exceptions.printStackTrace(ex);
-        }
-        engine.getContext().getBindings(ScriptContext.ENGINE_SCOPE).put("IO", System.out);
-        initFilters();
-        manager = new ExplorerManager();
-        manager.setRootContext(new AbstractNode(new FilterChildren()));
-        associateLookup(ExplorerUtils.createLookup(manager, getActionMap()));
-        view = new CheckListView();
-
-        ToolbarPool.getDefault().setPreferredIconSize(16);
-        Toolbar toolBar = new Toolbar();
-        toolBar.setBorder((Border) UIManager.get("Nb.Editor.Toolbar.border")); //NOI18N
-        toolBar.setMinimumSize(new Dimension(0,0)); // MacOS BUG with ToolbarWithOverflow
-
-        comboBox = new JComboBox();
-        toolBar.add(comboBox);
-        this.add(toolBar, BorderLayout.NORTH);
-        toolBar.add(SaveFilterSettingsAction.get(SaveFilterSettingsAction.class));
-        toolBar.add(RemoveFilterSettingsAction.get(RemoveFilterSettingsAction.class));
-        toolBar.addSeparator();
-        toolBar.add(NewFilterAction.get(NewFilterAction.class));
-        toolBar.add(RemoveFilterAction.get(RemoveFilterAction.class).createContextAwareInstance(this.getLookup()));
-        toolBar.add(MoveFilterUpAction.get(MoveFilterUpAction.class).createContextAwareInstance(this.getLookup()));
-        toolBar.add(MoveFilterDownAction.get(MoveFilterDownAction.class).createContextAwareInstance(this.getLookup()));
-        this.add(view, BorderLayout.CENTER);
-
-        filterSettings = new ArrayList<>();
-        updateComboBox();
-
-        comboBox.addActionListener(comboBoxActionListener);
-        setChain(filterChain);
-    }
-
     public void newFilter() {
         CustomFilter cf = new CustomFilter("My custom filter", "", engine);
         if (cf.openInEditor()) {
-            sequence.addFilter(cf);
-            FileObject fo = getFileObject(cf);
-            FilterChangedListener listener = new FilterChangedListener(fo, cf);
-            listener.changed(cf);
-            cf.getChangedEvent().addListener(listener);
+            addFilter(cf);
         }
     }
 
-    public void removeFilter(Filter f) {
-        com.sun.hotspot.igv.filter.CustomFilter cf = (com.sun.hotspot.igv.filter.CustomFilter) f;
+    public void addFilter(CustomFilter cf) {
```

Comment:
While at it, you could rename `cf` and `fo` to something more descriptive like `customFilter` and `fileObject`.

Reply:
done